src/jailer/src/env.rs

```diff
@@ -253,6 +250,17 @@ impl Env {
                 .into_empty_result()
                 .map_err(Error::CloseNetNsFd)?;
         }
+        Ok(())
+    }
+
+    pub fn run(mut self) -> Result<()> {
+        self.prepare_for_chroot()?;
+
+        // Safe to unwrap because we check in the call above that `file_name()` is successful.
+        // The borrow checker doesn't let us to nicely use an `&OsStr` as function parameter
```

Comment:
`OsStr` is a borrowed reference to an `OsString`from inside `self`. That is why the `self` is seen as mutably borrowed by the checker. What you can do is make `prepare_for_chroot `return an `OsString`. Something around:
```
fn prepare_for_chroot(&mut self) -> Result<(OsString)> {
...
Ok(exec_file_name.to_os_string())
}
```

```
pub fn run {
let exec_file_name = self.prepare_for_chroot()?;
let chroot_exec_file = PathBuf::from("/").join(&exec_file_name);

let cgroup = Cgroup::new(&self.id, self.numa_node, &exec_file_name)?;
}
```
WDYT?

Reply:
Nice! Indeed I tried something similar, but with `OsStr` instead of `OsString`:woman_facepalming:.
I'll update, thanks!